mobile/Ihc/services/LocalDataService.js

```diff
@@ -230,16 +233,37 @@ export function updateMedication(update) {
       properties.forEach( p => {
         medication[p] = update[p];
       });
-      return;
+      medication.key = Medication.makeKey(update);
+      return true;
     }
     //Medication does not exist (create)
+    update.key = Medication.makeKey(update);
     realm.create('Medication', update);
+    return true;
   });
+  return false;
+}
+
+// Returns the medication with the given name, dosage, and units; undefined if none found
+export function getMedication(drugName, dosage, units) {
+  const medication = realm.objects('Medication').filtered('drugName = "' + drugName +
+    '" AND dosage = "' + dosage + '" AND units = "' + units + '"');
+  return medication;
+}
+
+export function getMedicationWithKey(medicationKey) {
+  const medication = realm.objects('Medication').filtered('key = "' + medicationKey + '"');
```

Comment:
Like I said above, just return the exact medication here (extract the element from the array)